src/main/java/com/googlecode/protobuf/format/JsonFormat.java

```diff
@@ -875,6 +885,36 @@ protected void mergeField(Tokenizer tokenizer,
             mergeField(tokenizer, extensionRegistry, builder);
         }
     }
+    
+    private void handleUnknownField(Tokenizer tokenizer,    										
+    		Message.Builder builder,
+    		int fieldNumber) throws ParseException {    	
+    	tokenizer.tryConsume(":");
+    	if ("{".equals(tokenizer.currentToken())) {
+    		// Message structure
+    		tokenizer.consume("{");
+    		do {
+    			tokenizer.consumeIdentifier();
+    			handleUnknownField(tokenizer, builder, fieldNumber);

```

Comment:
I'm pretty fuzzy on how the unknown fields are handled - could there be an unknown field that has a value of a message with more unknown fields?  This recursion doesn't seem to build sub-messages.